subworkflows/local/clean_reads.nf

```diff
@@ -15,7 +15,37 @@ include { SEQTK_SAMPLE } from '../../modules/local/seqtk_sample.nf'
 
 
 
+process PUBLISH_FINAL_READS {
```

Comment:
I have tried thinking of a few ways to resolve this. The addition of a `final` tag was made by many others but I could not identify a clear way to avoid this issue of copying things. 

To elaborate on some of the difficulties in resolving the process that is `final` results from the pipeline not knowing what the `final` output is. As processes can be turned off or on depending on what happens.

Reply:
Thanks. That makes sense. Specifically, that there is more than one pathway/process that is used to manipulate the files, and so it's not known until runtime which of these processes is the `final` version of the files.

I did try to think of other ways to handle this too. But I can't think of any that don't overly complicate things. Since files are stored in Azure blobstorage, you can't just rename the files at the very end, you have to make a copy with a new name, as you are doing now. So the solution you have makes sense to me.